python/src/fragment.rs

```diff
@@ -41,6 +45,41 @@ impl FileFragment {
         format!("LanceFileFragment(id={})", self.fragment.id())
     }
 
+    #[staticmethod]
+    #[pyo3(signature = (dataset_uri, fragment_id, reader, **kwargs))]
+    fn create(
+        dataset_uri: &str,
+        fragment_id: usize,
+        reader: &PyAny,
+        kwargs: Option<&PyDict>,
+    ) -> PyResult<FragmentMetadata> {
+        let rt = tokio::runtime::Runtime::new()?;
+        let metadata = rt.block_on(async {
+            let mut batches: Box<dyn RecordBatchReader> = if reader.is_instance_of::<Scanner>()? {
+                let scanner: Scanner = reader.extract()?;
```

Comment:
have you tested this case in the if else here?

Reply:
added one test.